tests/integrations/gcp/test_gcp.py

```diff
@@ -0,0 +1,320 @@
+"""
+# GCP Cloud Functions system tests
+
+"""
+import json
+import os
+import time
+from textwrap import dedent
+import requests
+import uuid
+import tempfile
+import shutil
+import sys
+import subprocess
+
+import pytest
+import pickle
+import os.path
+import os
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+
+SCOPES = [
+    "https://www.googleapis.com/auth/cloud-platform",
+    "https://www.googleapis.com/auth/cloud-platform.read-only",
+    "https://www.googleapis.com/auth/cloudfunctions",
+    "https://www.googleapis.com/auth/logging.read",
+    "https://www.googleapis.com/auth/logging.admin",
+]
+
+FUNCTIONS_PRELUDE = """
+import sentry_sdk
+from sentry_sdk.integrations.gcp import GcpIntegration
+import json
+import time
+
+from sentry_sdk.transport import HttpTransport
+
+def event_processor(event):
+    # Adding delay which would allow us to capture events.
+    time.sleep(1)
+    return event
+
+class TestTransport(HttpTransport):
+    def _send_event(self, event):
+        event = event_processor(event)
+        # Writing a single string to stdout holds the GIL (seems like) and
+        # therefore cannot be interleaved with other threads. This is why we
+        # explicitly add a newline at the end even though `print` would provide
+        # us one.
+        print("\\nEVENTS: {}\\n".format(json.dumps(event)))
+
+def init_sdk(timeout_warning=False, **extra_init_args):
+    sentry_sdk.init(
+        dsn="https://123abc@example.com/123",
+        transport=TestTransport,
+        integrations=[GcpIntegration()],
+        shutdown_timeout=10,
+        **extra_init_args
+    )
+"""
+
+
+@pytest.fixture
+def authorized_credentials():
+    credentials = None
+
+    # Skipping tests if environment variables not set.
+    if "SENTRY_PYTHON_TEST_GCP_CREDENTIALS_JSON" not in os.environ:
+        pytest.skip("GCP environ vars not set")
+
+    # The file token.pickle stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first
+    # time.
+    with open(
+        os.environ.get("SENTRY_PYTHON_TEST_GCP_CREDENTIALS_JSON"), "rb"
+    ) as creds_file:
+        for line in creds_file.readlines():
```

Comment:
We will not be able to store any files in CI. Can we store the entire JSON in an envvar?

Also do I read it correctly that you're expecting that file to contain multiple JSON objects where you're only interested in the last one? Because you're splitting by line separator and override `creds_json` in each iteration.

Reply:
Actually it is only a single json object which is there in the file I'm just reading the file line by line and fetching one single json. 
The thing is that, these oauth client credentials are downloaded from  as a .json file. I was just setting this environment variable to that json which is stored in the file.
This will be a minor overhead on user to copy the json from the file and set it to the environment variable. Do we want to do that? If so, I'll modify this implementation.

Reply:
My main point was that it's easy to set secrets in Travis using environment variables but I may have misunderstood the auth flow.

Reading your other comment is there even a way to run this in CI if there's manual user interaction involved?

Reply:
I'm not completely sure about this, though what I can do is try to give all permissions from my google account to the APIs which are using the credentials. But I think for the first run atleast it still requires user interaction.